docs/topic-guides/blendable-web-apps/view-model-attaching.md

```diff
@@ -64,7 +64,21 @@ Handlers with empty tokens are called with other handlers with the same empty to
 
 ## Contexts
 
-Token matching can be made more fine-grained by using contexts. They are composed of a list of strings that acts as a bit map when matched with other contexts. No context \(`null` value\) means **match any context**. Otherwise, two handlers are matched if source context contains same elements as destination context. Examples:
+Token matching can be made more fine-grained by using contexts. They are composed of a list of strings that acts as a bit map when matched with other contexts. This list isn't materialized anywhere, it is just an un-written contract between the apps.
+
+Starcounter came up with few predefined contexts that have shared meaning which can be understood by all app authors:
+* `page` — an entire page content;
+* `thumbnail` — a thumbnail of a page content;
+* `row` — single row of entity data;
+* `icon` — launch icons or smaller icons in other scenarios;
+* `search` — a search result;
+* `app` — a meta context that represents the current app;
```

Comment:
That's very good but it inspired me to try to make it even clearer:

* `page` — a size factor context. Use it if the view contains full information that the app has about a concept. The view renders in multiple lines. The view is suitable to be attached in full pages about a concept;
* `thumbnail` — a size factor context. Use it if the view contains basic information that the app has about a concept. The view renders in multiple lines. The view is suitable to be attached in side information or teasers about a concept;
* `row` — a size factor context. Use it if the view contains basic information that the app has about a concept. The view renders in a single line. The view is suitable to be attached in lists;
* `icon` — a size factor context. Use it if the view contains the smallest unit of information that the app has about a concept. The view renders in a single element that is a link to a bigger view. The view is suitable to be attached in menu bars;
* `search` — a meta context. Use it if the view is suitable for search results;
* `app` — a meta context. Use it if the view represents the app itself, rather than a concept;

There are some compound contexts, that have a meaning of their own:

* `search`, `row` — a single-line row that is suitable for search results;
* `app`, `icon` — an icon that opens the main page of the app;